### PR TITLE
[#187364713] Configuration of Postgres and Sequelise

### PR DESCRIPTION
**Pull request by mpanoathos**
**What does this PR do?**
Has configurations of postgres and sequelise and some endpoints to check whether the database is working

**Have the following end-points working**

- GET/welcome
- GET /users
- POST /users

**How should this be tested**
By using postman

**Pivotal tracker story**
#187364713

### CODE REVIEW

**Review by AngeloChristian1 — APPROVED** [review]

**Review by Yvan-David — APPROVED** [review]

File: index.ts [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
-    console.log(`listening on port https://localhost:${port}`)
-}) 
+//Testing if the database is authenticated and listening the port
+async function startServer() {
+    try {
+        await sequelize.authenticate();
+        console.log('Connection to the database has been established successfully.');
+
+        await sequelize.sync();
+        console.log('All models were synchronized successfully.');
```
Comment (munezerobagira) [inline index.ts]:
```suggestion

```
We will only use migrations, syncing database can cause unexpected loss of data

File: src/routes/route.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  try {
+    const { name, email, password } = req.body;
+    // Create the user
+    const user = await User.create({ name, email, password });
+    res.status(201).json(user);
+  } catch (error) {
+    handleError(res, error);
+  }
+});
+
```
Comment (munezerobagira) [inline src/routes/route.ts]:
```suggestion

```
only one sample get request is enough

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/migrations/migrations.ts
Comment (munezerobagira) [inline src/migrations/migrations.ts:1]: Please remove sequelize.sync entirely

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by munezerobagira — APPROVED** [review]

File: src/routes/route.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  try {
+    // Retrieve all users from the database
+    const users = await UserModel.findAll();
+    res.json(users);
+  } catch (error) {
+    handleError(res, error);
+  }
+});
+
+export default router;
```
Comment (munezerobagira) [inline src/routes/route.ts]: this section need to be tested

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -2,9 +2,9 @@
   "name": "e-commerce-bitcrafters-bn",
   "version": "1.0.0",
   "description": "",
-  "main": "dist/index.js",
+  "main": "index.ts",
   "scripts": {
-    "start": "node dist/index.js",
+    "start": "nodemon",
```
Comment (munezerobagira) [inline package.json]: why did change this line?

File: package.json [hunk 5]
```diff
@@ -2,9 +2,9 @@
   "name": "e-commerce-bitcrafters-bn",
   "version": "1.0.0",
   "description": "",
-  "main": "dist/index.js",
```
Comment (munezerobagira) [inline package.json]: why did you also change this line

File: tsconfig.json [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
+    // "moduleResolution":"node",
+    "target": "ES6",
     "baseUrl": "src",
-    "outDir": "./dist",
-    "esModuleInterop": true,
+    "outDir": "dist",
     "sourceMap": true,
-    "noImplicitAny": true,
-    "allowJs": true
+    "noImplicitAny": true
```
Comment (munezerobagira) [inline tsconfig.json]: why did you change this file?

Comment (mpanoathos) [reply tsconfig.json]: It was due to the errors I had before, but now I have corrected it

File: package.json [hunk 7]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "dist/index.js",
   "scripts": {
-    "start": "node dist/index.js",
+    "start": "node dist/nodemon",
```
Comment (munezerobagira) [inline package.json]:
```suggestion
    "start": "node dist/index.js",
```

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by munezerobagira — APPROVED** [review]
